src/core/control/tools/StrokeHandler.cpp

```diff
@@ -32,26 +32,24 @@ StrokeHandler::~StrokeHandler() {
 }
 
 void StrokeHandler::draw(cairo_t* cr) {
-    if (!stroke) {
-        return;
-    }
-
-    if (this->fullRedraw) {
-        /**
-         * Erase the mask entirely in this case
-         */
-        cairo_set_operator(crMask, CAIRO_OPERATOR_CLEAR);
-        cairo_paint(crMask);
+    assert(stroke && stroke->getPointCount() > 0);
```

Comment:
I'm conflicted about this assertion. If I understand correctly, on one hand, we want to avoid having to draw a stroke if the user has not initiated stroke input. On the other hand, it does not seem obvious to me that draw is only invoked in the above situation, which means that a user could get segfaults when drawing stuff.

The assertion can be improved as follows:
```suggestion
    g_assert(stroke && stroke->getPointCount() > 0 && "can only draw a stroke if user has initiated stroke input!");
```

Reply:
I don't think `draw` can ever be called if the stroke has not yet been created. The stroke is created, with an initial point, at the same time as the StrokeHandler is created (see `XojPageView::onButtonPressEvent`).